internal/window/window.go

```diff
@@ -34,17 +34,16 @@ type Window [WindowSliceSize]byte
 // [4, 0, 0, 0]
 // Roll >= 4
 // [0, 0, 0, 0]
-// Assumes that [roll] is greater than or equal to 0
-func Roll(w Window, roll int64) (Window, error) {
+func Roll(w Window, roll uint64) Window {
 	// Note: make allocates a zeroed array, so we are guaranteed
 	// that what we do not copy into, will be set to 0
 	res := [WindowSliceSize]byte{}
 	bound := roll * consts.Uint64Len
 	if bound > WindowSliceSize {
-		return res, nil
+		return res
```

Comment:
this change would miss the overflow protection. to fix it:
```golang
if bound >= WindowSliceSize  || roll >= WindowSliceSize {
```
( note the `>=` on the bound side, as it would prevent a potential oob )